File: 75. Sort Colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int n = nums.size();
        int i = 0;
        while (i < n)
        {
            if (nums[i] == 0)
            {
                int temp = nums[i];
                nums.erase(nums.begin() + i);
                if (nums.size() > 0)
                    nums.insert(nums.begin(), temp);
                else
                    nums.push_back(temp);
            }
            else if (nums[i] == 2)
            {
                int temp = nums[i];
                nums.erase(nums.begin() + i);
                nums.push_back(temp);
                i--;
                n--;
            }
            i++;
        }
    }
};

File: 75. Sort Colors.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        i = 0
        n = len(nums)
        while i < n:
            if nums[i] == 0:
                temp = nums.pop(i)
                if nums:
                    nums.insert(0,temp)
                else:
                    nums += [temp]
            elif nums[i] == 2:
                temp = nums.pop(i)
                nums.append(temp)
                i -= 1
                n -= 1
            i += 1
